### ISSUE TITLE
`nix` ui: list and remove gc roots

### ISSUE THREAD

**Issue by FRidh** [NixOS/nix#4633]
**Is your feature request related to a problem? Please describe.**
Quite often I run out of disk space because of gcroots that I don't need. To find the roots, I now use
```
$ ls -l $(readlink /nix/var/nix/gcroots/auto/*)
```
and remove them when I think they can be removed.

**Describe the solution you'd like**
I'd like the `nix` command to show me the list of roots, where they point to, when they were created and how much space could/would be freed when removing it.

**Describe alternatives you've considered**
The snippet I listed above. Or maybe a more generic garbage collection command.

**Comment by psionic-k** [comment]
Yeah.  We could use a frontend for this.  Checklist I just went through:

https://www.reddit.com/r/NixOS/comments/1l2u3tx/reminder_delete_old_roots/

- home manager generations have roots
- every project's direnv can have roots
- random result and result-bin roots

Just went from about 280GB to 66GB in the store and still more reduction to come.  It's not easy to see which roots are soaking up the space.  We need a consultative workflow that shows us where the weight is and gives us a chance to delete old and heavy roots first.
